src/wallet/tx_builder.rs

```diff
@@ -542,6 +543,15 @@ impl<'a, D: BatchDatabase, Cs: CoinSelectionAlgorithm<D>, Ctx: TxBuilderContext>
         self.params.rbf = Some(RbfValue::Value(nsequence));
         self
     }
+
+    /// Set the current blockchain height.
+    ///
+    /// This will be used to set the nLockTime for preventing fee sniping. If the current height is
+    /// not provided, the last sync height will be used instead.
+    pub fn set_current_height(&mut self, height: BlockTime) -> &mut Self {
```

Comment:
I would use `u32` so that we don't have to create a `BlockTime` struct with an unused timestamp